### DISCUSSION TITLE
How to set up working directory for site-map notebook

### DISCUSSION THREAD

**Discussion by advythr** [earthlab-education/Earth-Analytics-2023-01-Intro#499] [Q&A06 Vector Data]
I am running a Jupyter notebook in GitHub Codespaces configured as 8-core.

I'm trying to set up my working directory, download my data, and put it into the working directory so I can open the file 

> california/madera-county-roads/tl_2013_06039_roads.shp

 in the next part of the notebook **Open And Clip Your Vector Data**.

The following code seems to correctly set up my working directory:

```python
# Import packages, download data, and set working directory here
import geopandas as gpd
import holoviews as hv
import earthpy as et
import os
import pathlib

spatial_vector = et.data.get_data('spatial-vector-lidar')
spatial_vector

# Generating the path to the project directory
project_dir = os.path.join(
    pathlib.Path.home(), "earth-analytics", "data", "site-map"
)

# Create the project directory
os.makedirs(project_dir, exist_ok=True)
project_dir
```

But when I try to open the file using this code:
```Python
madera_roads_path = "site-map/california/madera-county-roads/tl_2013_06039_roads.shp"


# Open the geospatial file
madera_roads_gdf = gpd.read_file(madera_roads_path)
```

I get the error: 

> DriverError: site-map/california/madera-county-roads/tl_2013_06039_roads.shp: No such file or directory

What am I doing wrong here? I suspect I didn't put the downloaded data into the right folder, but I'm not sure.

**Comment by eculler** [comment]
Excellent post! It's very clear to me what the problem is and what might fix it.

There's a couple things going on here:
1. earthpy *always* downloads this dataset to `~/earth-analytics-/data/spatial-vector-lidar` and *not* `~/earth-analytics-/data/site-map`. Another way of thinking about it is that the project directory in this case is already created by `earthpy`, so you don't need to do it yourself.
2. You are not changing your working directory, but you are using a relative path. You should either set the working directory or use absolute paths
3. You're not using `os.path.join()` - this is important in case someone tries to run your code on a Windows computer!

**Reply by eculler** [reply]
Two ways to approach this:

## Absolute paths, using the output of `et.data.get_data` to get the download directory
```python
madera_roads_path = os.path.join(
    spatial_vector, 
    "california", "madera-county-roads", "tl_2013_06039_roads.shp")
```

## Relative paths and changing the working directory
```python
os.chdir(spatial_vector)
madera_roads_path = os.path.join(
    "california", "madera-county-roads", "tl_2013_06039_roads.shp"))
```

**Reply by edchanonline** [reply]
My approach
```
os.chdir(natural_earth_path)
madera_roads_gdf = gpd.read_file('./california/madera-county-roads/tl_2013_06039_roads.shp')
neon_sjer_gdf = gpd.read_file('./california/neon-sjer-site/vector_data/SJER_crop.shp')
```

Food for thought!

**Comment by taylor-obrien** [comment]
Another way to check if your data is where you think it is, is by going into the terminal. In the terminal, you can change directory or see where you are in a directory. I will show you in a screenshot below. cd will change directory and ls will show you what is located in the directory you are currently in. Might be a good thing to know for the future, for when you are confused on where you should be pulling your data from. Good luck!

![image](https://github.com/earthlab-education/Earth-Analytics-2023-01-Intro/assets/143227722/34f796dc-6a0e-438c-a84d-22bd2de5b420)
